torchgeo/datasets/spacenet.py

```diff
@@ -177,43 +190,32 @@ def _load_mask(
         try:
             with fiona.open(path) as src:
                 vector_crs = CRS(src.crs)
-                if raster_crs == vector_crs:
-                    labels = [feature['geometry'] for feature in src]
-                else:
-                    labels = [
-                        transform_geom(
-                            vector_crs.to_string(),
-                            raster_crs.to_string(),
-                            feature['geometry'],
-                        )
-                        for feature in src
-                    ]
-        except FionaValueError:
+                labels = [
+                    transform_geom(
+                        vector_crs.to_string(),
+                        raster_crs.to_string(),
+                        feature['geometry'],
+                    )
+                    for feature in src
+                    if feature['geometry']
+                ]
+        except (FionaError, FionaValueError):
+            # Empty geojson files, geometries that cannot be transformed (SN7)
             labels = []
 
-        if not labels:
-            mask_data = np.zeros(shape=shape)
-        else:
-            mask_data = rasterize(
+        if labels:
+            mask = rasterize(
                 labels,
                 out_shape=shape,
                 fill=0,  # nodata value
                 transform=tfm,
                 all_touched=False,
-                dtype=np.uint8,
+                dtype=np.int64,
```

Comment:
why int64 here?

Reply:
Instead of rasterizing to uint8 and casting to int64 (aka `torch.long`), we now just rasterize directly to int64.